### PR TITLE
Fix undefined behavior sanitizer issues in RTDB

### LINKED ISSUE #1443: Invalid Shift
https://github.com/firebase/firebase-ios-sdk/blob/137ce37363d72ac1d643ea9f93bc6e6665367e57/Firebase/Database/Api/FIRDatabaseQuery.m#L256

Left shift of 1 by 31 places cannot be represented in type 'long'

https://developer.apple.com/documentation/code_diagnostics/undefined_behavior_sanitizer/invalid_shift

### LINKED ISSUE #1444: Invalid bool load
https://github.com/firebase/firebase-ios-sdk/blob/31a2acb7497eaaeae38b7ceb17edcab09d7f2866/Firebase/Database/Snapshot/FIndexedNode.m#L82

https://developer.apple.com/documentation/code_diagnostics/undefined_behavior_sanitizer/invalid_boolean

### PR DESCRIPTION
**Pull request by paulb777**
Fix #1443. Fix #1444.

### CODE REVIEW

File: Firebase/Database/Persistence/FLevelDBStorageEngine.m [hunk 1]
```diff
@@ -672,7 +672,7 @@ - (NSData*) serializePrimitive:(id)value {
     return [data subdataWithRange:NSMakeRange(1, data.length - 2)];
 }
 
-- (id)fixDoubleParsing:(id)value {
+- (id)fixDoubleParsing:(id)value __attribute__((no_sanitize("float-cast-overflow"))) {
```
Comment (wilhuff) [inline Firebase/Database/Persistence/FLevelDBStorageEngine.m]: A better fix here would be to cast through an `int64_t` rather than `long long`.

Comment (wilhuff) [reply Firebase/Database/Persistence/FLevelDBStorageEngine.m]: Sorry I wasn't clear: I meant while removing the `__attribute__` you added. Casting through `int64_t` guarantees that the type is large enough to hold a `double` so the overflow becomes impossible.

Comment (paulb777) [reply Firebase/Database/Persistence/FLevelDBStorageEngine.m]: @wilhuff This change doesn't fix the sanitizer issue.  I believe that for any architecture we care about `long long` is always the same as `int64_t`. The undefined behavior is from casting the `int64_t` back to a `double`, not from the cast to the `int64_t`.

Comment (paulb777) [reply Firebase/Database/Persistence/FLevelDBStorageEngine.m]: screen shot 2018-06-25 at 9 13 41 am

**Pull request merged**